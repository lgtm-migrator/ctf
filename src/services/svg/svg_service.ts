// https://twitter.com/LeaVerou/status/1241619866475474946

// https://stackoverflow.com/a/28692538/725901
export const content = (emoji: string, size = 256) =>
  `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 ${size} ${size}" width="${size}" height="${size}">
  <text y=".85em" font-size="${size}">${emoji}</text>
</svg>
`;

export function toDataURI(content: string) {
  return (
    'data:image/svg+xml;base64,' + btoa(unescape(encodeURIComponent(content)))
  );
}
