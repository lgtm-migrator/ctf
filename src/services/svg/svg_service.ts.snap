// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVG service when producing content for a country flag emoji should output valid XML 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 128 128\\" width=\\"128\\" height=\\"128\\">
  <text y=\\".85em\\" font-size=\\"128\\">üá©üá∞</text>
</svg>"
`;

exports[`SVG service when producing content for a flag emoji should output valid XML 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 128 128\\" width=\\"128\\" height=\\"128\\">
  <text y=\\".85em\\" font-size=\\"128\\">üèÅ</text>
</svg>"
`;

exports[`SVG service when producing content for an emoji should output valid XML 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 128 128\\" width=\\"128\\" height=\\"128\\">
  <text y=\\".85em\\" font-size=\\"128\\">üôÉ</text>
</svg>"
`;

exports[`SVG service when producing data URI with emoji üá©üá∞ of size 16px should generate expected URI 1`] = `"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxNiAxNiIgd2lkdGg9IjE2IiBoZWlnaHQ9IjE2Ij4KICA8dGV4dCB5PSIuODVlbSIgZm9udC1zaXplPSIxNiI+8J+HqfCfh7A8L3RleHQ+Cjwvc3ZnPgo="`;

exports[`SVG service when producing data URI with emoji üá©üá∞ of size 32px should generate expected URI 1`] = `"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAzMiAzMiIgd2lkdGg9IjMyIiBoZWlnaHQ9IjMyIj4KICA8dGV4dCB5PSIuODVlbSIgZm9udC1zaXplPSIzMiI+8J+HqfCfh7A8L3RleHQ+Cjwvc3ZnPgo="`;

exports[`SVG service when producing data URI with emoji üá©üá∞ of size 64px should generate expected URI 1`] = `"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA2NCA2NCIgd2lkdGg9IjY0IiBoZWlnaHQ9IjY0Ij4KICA8dGV4dCB5PSIuODVlbSIgZm9udC1zaXplPSI2NCI+8J+HqfCfh7A8L3RleHQ+Cjwvc3ZnPgo="`;

exports[`SVG service when producing data URI with emoji üèÅ of size 16px should generate expected URI 1`] = `"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxNiAxNiIgd2lkdGg9IjE2IiBoZWlnaHQ9IjE2Ij4KICA8dGV4dCB5PSIuODVlbSIgZm9udC1zaXplPSIxNiI+8J+PgTwvdGV4dD4KPC9zdmc+Cg=="`;

exports[`SVG service when producing data URI with emoji üèÅ of size 32px should generate expected URI 1`] = `"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAzMiAzMiIgd2lkdGg9IjMyIiBoZWlnaHQ9IjMyIj4KICA8dGV4dCB5PSIuODVlbSIgZm9udC1zaXplPSIzMiI+8J+PgTwvdGV4dD4KPC9zdmc+Cg=="`;

exports[`SVG service when producing data URI with emoji üèÅ of size 64px should generate expected URI 1`] = `"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA2NCA2NCIgd2lkdGg9IjY0IiBoZWlnaHQ9IjY0Ij4KICA8dGV4dCB5PSIuODVlbSIgZm9udC1zaXplPSI2NCI+8J+PgTwvdGV4dD4KPC9zdmc+Cg=="`;

exports[`SVG service when producing data URI with emoji üôÉ of size 16px should generate expected URI 1`] = `"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxNiAxNiIgd2lkdGg9IjE2IiBoZWlnaHQ9IjE2Ij4KICA8dGV4dCB5PSIuODVlbSIgZm9udC1zaXplPSIxNiI+8J+ZgzwvdGV4dD4KPC9zdmc+Cg=="`;

exports[`SVG service when producing data URI with emoji üôÉ of size 32px should generate expected URI 1`] = `"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAzMiAzMiIgd2lkdGg9IjMyIiBoZWlnaHQ9IjMyIj4KICA8dGV4dCB5PSIuODVlbSIgZm9udC1zaXplPSIzMiI+8J+ZgzwvdGV4dD4KPC9zdmc+Cg=="`;

exports[`SVG service when producing data URI with emoji üôÉ of size 64px should generate expected URI 1`] = `"data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA2NCA2NCIgd2lkdGg9IjY0IiBoZWlnaHQ9IjY0Ij4KICA8dGV4dCB5PSIuODVlbSIgZm9udC1zaXplPSI2NCI+8J+ZgzwvdGV4dD4KPC9zdmc+Cg=="`;
